pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -137,7 +196,13 @@ func (r *ReconcileArtifactBuildRequest) handleStateDiscovered(ctx context.Contex
 		}
 	} else {
 		//build already exists, add us to the owner references
+		var recent *v1alpha1.DependencyBuild
 		for _, db := range list.Items {
+			//there should only ever be one build
+			//if somehow we do end up with two we consider the most recent one
+			if recent == nil || recent.CreationTimestamp.Before(&db.CreationTimestamp) {
```

Comment:
I wonder it this situation would also warrant an event for debug purposes

let's put a TODO comment to at least think about logging or posting an event